components/provisioner/assets/database/provisioner.sql

```diff
@@ -0,0 +1,95 @@
+-- Cluster Config
+
+CREATE TABLE "GardenerConfig"
+(
+    id uuid PRIMARY KEY CHECK (id <> '00000000-0000-0000-0000-000000000000'),
+    name varchar(256) NOT NULL,
+    kubernetesVersion varchar(256) NOT NULL,
+    nodeCount integer NOT NULL,
+    volumeSize varchar(256) NOT NULL,
+    machineType varchar(256) NOT NULL,
+    region varchar(256) NOT NULL,
+    targetProvider varchar(256) NOT NULL,
+    targetSecret varchar(256) NOT NULL,
+    zone varchar(256) NOT NULL,
+    cidr varchar(256) NOT NULL,
+    autoScalerMin integer NOT NULL,
+    autoScalerMax integer NOT NULL,
+    maxSurge integer NOT NULL,
+    maxUnavailable integer NOT NULL
+);
+
+CREATE TABLE "GCPConfig"
+(
+    id uuid PRIMARY KEY CHECK (id <> '00000000-0000-0000-0000-000000000000'),
+    name varchar(256) NOT NULL,
+    kubernetesVersion varchar(256) NOT NULL,
+    numberOfNodes integer NOT NULL,
+    bootDiskSize varchar(256) NOT NULL,
+    machineType varchar(256) NOT NULL,
+    region varchar(256) NOT NULL,
+    zone varchar(256) NOT NULL
+);
+
+
+-- Cluster
+
+CREATE TABLE "Cluster"
+(
+    id uuid PRIMARY KEY CHECK (id <> '00000000-0000-0000-0000-000000000000'),
+    kubeconfig text,
+    terraform_state jsonb,
+    creation_timestamp timestamp without time zone NOT NULL,
+    gardener_config_id uuid,
+    gcp_config_id uuid,
+    foreign key (gardener_config_id) REFERENCES "GardenerConfig" (id) ON DELETE CASCADE,
```

Comment:
After some consideration I think we should change the direction of the relation.
We have the following relations in the current design:
- Cluster have Gardener Configuration with cardinality 0:1
- Cluster have GCP Configuration with cardinality 0:1
- Kyma Configuration is related to Cluster with cardinality 1:1

We can change the design as follows:
- Gardener Configuration is related to Cluster with cardinality 1:1
- GCP Configuration is related to Cluster with cardinality 1:1
- Kyma Configuration is related to Cluster with cardinality 1:1

With the current design we need to create records in the following order:
- Gardener/GCP Configuration table
- Cluster table
- Kyma Configuration
- Operation

In case we add ClusterID field to KymaConfig table the order will be slightly changed:
- Cluster table
- Gardener/GCP Configuration table
- Kyma Configuration
- Operation

The consequence of this change is that before querying for runtime data we need to check what is the type of configuration. Having fields with configuration ids as in the current design is slightly easier (if we change the design as proposed there is a need to do a query on both GCPConfig and GardenerConfig tables)